src/pyspiffe/workloadapi/default_workload_api_client.py

```diff
@@ -34,6 +37,69 @@
     WORKLOAD_API_HEADER_VALUE,
 )
 
+__all__ = ['DefaultWorkloadApiClient']
+
+# GRPC Error Codes that the client will not retry on:
+#  - INVALID_ARGUMENT is not retried according to the SPIFFE spec because the request is invalid
+#  - CANCELLED is not retried because it occurs when the caller has canceled the operation.
+_NON_RETRYABLE_CODES = {grpc.StatusCode.CANCELLED, grpc.StatusCode.INVALID_ARGUMENT}
+
+
+class RetryHandler:
+    """Handler that performs retries using an exponential backoff policy."""
+
+    UNLIMITED_RETRIES = 0
+
+    def __init__(
+        self,
+        max_retries: int = UNLIMITED_RETRIES,
```

Comment:
I was wondering if 0 should mean no retry at all and UNLIMITES_RETRIES should be -1.

Reply:
We don't need to codify _no retry_ in the RetryHandler since we would never create it in case we don't want retries.

Reply:
...and I read it as max-retries = 0 = False => no max at all, as in python _0_ evaluates as _False_

Reply:
Makes sense.